frontend/templates/product-list/sections/FilterableProductsSection/RefinementSingleSelect.tsx

```diff
@@ -29,6 +31,17 @@ export function RefinementSingleSelect({
          ...otherProps,
          sortBy: useSortBy(otherProps),
       });
+
+   const onClickSingleSelect = React.useCallback(((newSelected: RefinementListItem) => {
+      refine(newSelected.value);
+
+      const oldSelected: RefinementListItem | undefined = items.find(
+         (item) => item.isRefined
+      );
+      if (oldSelected && oldSelected !== newSelected) {
+         refine(oldSelected.value);
+      }
+   }), [items]);
```

Comment:
Shouldn't `refine` be a part of the dependency array here? We don't want to use a stale version of the `refine` callback.

Reply:
how would a function be stale?

Reply:
oh cause its a prop i see

Reply:
i see we craft it out of some props it looks like so it could change potentially